test/box-luatest/gh_9266_fix_on_shutdown_and_osexit_and_cmd_expr_test.lua

```diff
@@ -0,0 +1,32 @@
+local t = require('luatest')
+local popen = require('popen')
+
+local g = t.group()
+
+local tarantool = arg[-1]
+
+g.after_each(function()
+    if g.handle ~= nil then
+        g.handle:close()
+    end
+    g.handle = nil
+end)
+
+g.test = function()
+    local code = [[
+        box.ctl.on_shutdown(function()
+            print('hello')
+        end)
+        os.exit()
+    ]]
+    local handle, err = popen.new({tarantool, '-e', code},
+                                  {stdout = popen.opts.PIPE})
```

Comment:
I would also set `stdin`/`stderr` to `popen.opts.DEVNULL`. If something goes bad and the child process remains alive, it keeps corresponding pipe ends open, so test-run may stuck waiting for it (at least for `stderr`).

Reply:
Done.

```diff
diff --git a/test/box-luatest/gh_9266_fix_on_shutdown_and_osexit_and_cmd_expr_test.lu>
index 01889a7e9..15152652a 100644
--- a/test/box-luatest/gh_9266_fix_on_shutdown_and_osexit_and_cmd_expr_test.lua
+++ b/test/box-luatest/gh_9266_fix_on_shutdown_and_osexit_and_cmd_expr_test.lua
@@ -20,7 +20,9 @@ g.test = function()
         os.exit()
     ]]
     local handle, err = popen.new({tarantool, '-e', code},
-                                  {stdout = popen.opts.PIPE})
+                                  {stdout = popen.opts.PIPE,
+                                   stdin = popen.opts.DEVNULL,
+                                   stderr = popen.opts.DEVNULL})
     assert(handle, err)
     g.handle = handle
     local output, err = handle:read({timeout = 3})
```